DSpace/dspace-angular
# Refactored community & collection pages

## References
* Fixes #2425
* Fixes #2700
* Requires #2339

## Description
The reason why the community/collection title remained was because the global browse by page and the community & collection browse pages used the same component (`BrowseByMetadataPageComponent`). When you went to a community/collection browse page the comcol header & footer that were hardcoded on the global browse page were shown, otherwise they were hidden. Because changing from scope didn't change the component that should be rendered (because both wit or without scope the global browse page was displayed), the page didn't refresh completely. This isn't really a clean way of creating tabs because the community & collection header & footer were duplicate. The cleanest way was to completely refactor the global browse pages and to extract the code that needed to be displayed on both the comcol pages and the global browse pages and move them into browse sections. These sections can then be displayed on the newly created `BrowseByPageComponent` or as a tab on the community/collection pages.

## Instructions for Reviewers
List of changes in this PR:
* Created the `SubComColSectionComponent` containing the community's `Subcommunities and Collections` tab content
* Created the `CollectionRecentlyAddedComponent` containing collection's the `Recent Submissions` tab content
* Removed the comcol header & footer from the `BrowseByMetadataPageComponent` because now this is not needed anymore
* Updated the community & collection routing modules in order to be able to work with the `app-routlet`
* Created a new tab `ComcolBrowseByComponent` that will display the browse sections for both community & collection pages
* Created a new page `BrowseByMetadataPageComponent` that will be used to render the global browse by pages by using the browse sections

During this PR I also found & fixed additional bugs:
- The browse by `dateissued` has an infinite load animation when there are 0 items to display
  -  Fixed this by creating a separate `loading$` variable to that is automatically upated when the backend sends a response back
- The hierarchical browse pages didn't support scoped searches
  - Fixed this by adding the scope to the search query when clicking on the `browse` link
- When you switch between 2 tabs who used the `ComcolBrowseByComponent` the rendered browse section weren't automatically updated. This was because the parameters (in this case `browseByType`) who are used to retrieve the component in `getComponent()` changed without calling the `getComponent` method again. Because the component wasn't reinitialised switching between 2 browse tabs didn't have any effect and it looked like you were still on the same page.
  - Because this is a general component loader bug I moved the fix to #2339 (instead of fixing it in all the loaders individually). This is also the reason why this PR is dependant from that PR
- Themed versions of the browse pages were created, but these are dynamically loaded components using custom decorators, so the correct way to theme those is using the theme parameter in `@rendersBrowseBy(BrowseByDataType, theme)`
  - Also moved this fix to #2339

**Guidance for how to test or review this PR:**
- Verify that the community & collection tabs work
- Verify that the browse sections on the community & collection pages don't display items from different collections
- Verify that the community & collection browse by `dateissued` tab correctly displays the message `No items to show.` instead of the infinite load animation
- Verify that the accessibility issues on the browse page are fixed
- Verify that the hierarchal browse tabs on the community & collection pages now also display their respective header & footer and that the browse button only displays results inside that scope
- Verify that the global hierarchal browse doesn't set a scope

## Checklist
